pkg/cmd/experimental/project/new_project.go

```diff
@@ -0,0 +1,117 @@
+	"github.com/openshift/origin/pkg/client"
+	projectapi "github.com/openshift/origin/pkg/project/api"
+)
+
+type newProjectOptions struct {
+	projectName  string
+	displayName  string
+	description  string
+	clientConfig clientcmd.ClientConfig
+
+	adminRole             string
+	masterPolicyNamespace string
+	adminUser             string
+}
+
+func NewCmdNewProject(clientConfig clientcmd.ClientConfig) *cobra.Command {
+	options := &newProjectOptions{clientConfig: clientConfig}
+
+	cmd := &cobra.Command{
+		Use:   "new-project <project-name>",
+		Short: "create a new project",
+		Long:  `create a new project`,
+		Run: func(cmd *cobra.Command, args []string) {
+			if !options.complete(cmd) {
+				return
+			}
+
+			err := options.run()
+			if err != nil {
+				fmt.Printf("%v\n", err)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&options.masterPolicyNamespace, "master-policy-namespace", "master", "master policy namespace")
+	cmd.Flags().StringVar(&options.adminRole, "admin-role", "admin", "project admin role name in the master policy namespace")
+	cmd.Flags().StringVar(&options.adminUser, "admin", "", "project admin username")
+	cmd.Flags().StringVar(&options.displayName, "display-name", "", "project display name.  defaults to project-name")
+	cmd.Flags().StringVar(&options.description, "description", "", "project description.  defaults to display-name")
+
+	return cmd
+}
+
+func (o *newProjectOptions) complete(cmd *cobra.Command) bool {
+	args := cmd.Flags().Args()
+	if len(args) != 1 {
+		cmd.Help()
+		return false
+	}
+
+	o.projectName = args[0]
+
+	if len(o.displayName) == 0 {
+		o.displayName = o.projectName
+	}
+
+	return true
+}
+
+func (o *newProjectOptions) run() error {
+	clientConfig, err := o.clientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+	client, err := client.New(clientConfig)
+	if err != nil {
+		return err
+	}
+
+	_, err = client.Projects().Get(o.projectName)
+	projectFound := !kerrors.IsNotFound(err)
+	if (err != nil) && (projectFound) {
+		return err
+	}
+	if projectFound {
+		return fmt.Errorf("project %v already exists", o.projectName)
+	}
+
+	project := &projectapi.Project{}
+	project.Name = o.projectName
+	project.DisplayName = o.displayName
+	project.Annotations = make(map[string]string)
+	project.Annotations["description"] = o.description
+	project, err = client.Projects().Create(project)
+	if err != nil {
+		return err
+	}
+
+	if len(o.adminUser) != 0 {
+		adminRoleBinding := &authorizationapi.RoleBinding{}
+
+		adminRoleBinding.Name = "admins"
+		adminRoleBinding.RoleRef.Namespace = o.masterPolicyNamespace
+		adminRoleBinding.RoleRef.Name = o.adminRole
+		adminRoleBinding.UserNames = []string{o.adminUser}
+
+		_, err := client.RoleBindings(project.Name).Create(adminRoleBinding)
+		if err != nil {
+			// try to delete the project we just created.  ignore any result, because we're failing anyway and this is best effort
+			client.Projects().Delete(project.Name)
```

Comment:
alternately, if we want this to be a more hand-holding command, we could just tell them how to add additional users (call out the `openshift ex add-user` command), or in the failure case, how to add a user to the project they just created